File: src/main/java/nnglebanov/daoexample/repositories/BookRepository.java
package nnglebanov.daoexample.repositories;

import nnglebanov.daoexample.domain.Book;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Transactional
public interface BookRepository extends JpaRepository<Book, Integer> {
    @Query("select b from Book b join b.comments c where c.commentText like %:comment%")
    List<Book> selectBooksWhithCommentLike(@Param("comment") String comment);
}

File: src/main/java/nnglebanov/daoexample/repositories/BookRepository.kt
package nnglebanov.daoexample.repositories

import nnglebanov.daoexample.domain.Book
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import org.springframework.stereotype.Repository
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono



@Repository
interface BookRepository : ReactiveMongoRepository<Book, String> {
    override fun findAll(): Flux<Book>

    override fun findById(id: String): Mono<Book>

    fun save(person: Mono<Book>): Mono<Book>
}